sdks/java/io/synthetic/src/test/java/org/apache/beam/sdk/io/synthetic/SyntheticStepTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.io.synthetic;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.transforms.DoFnTester;
+import org.apache.beam.sdk.values.KV;
+import org.apache.commons.math3.distribution.ConstantRealDistribution;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link SyntheticStep}. */
+@RunWith(JUnit4.class)
+public class SyntheticStepTest {
```

Comment:
I'm wondering: is preserving input key distribution the only thing we can/want to test here? 

SyntheticStep is quite known to me by now, but I have a feeling that a new developer that wants to learn about what it does, will look for tests as a source of truth (I do this very often). Obviously, other than that, more tests would check the code better. 

WDYT about testing other functionalities of SyntheticStep, eg, testng amplification/filtering, throughput, delay, bundle delay? I think we can test all this with TestPipeline on Direct runner. Please correct me if I'm wrong (you may have more prior experience with it).

Reply:
If this is not an easy thing to do, I think we can skip this in this PR and add tests later but we should file a JIRA for this.

Reply:
I can add a JIRA to ahve a good set of tests that exercise these classes. Do you think it makes sense to have them as unit tests with testpipeline on directrunner ?

Reply:
ADded https://issues.apache.org/jira/browse/BEAM-5188